extensions/amp-analytics/0.1/transport.js

```diff
@@ -118,13 +131,224 @@ export class Transport {
     xhr.send('');
     return true;
   }
+
+  /**
+   * Sends an Amp Analytics trigger event to a vendor's cross-domain iframe,
+   * or queues the message if the frame is not yet ready to receive messages.
+   * @param {string} request
+   * @param {!Object<string, string>} transportOptions
+   */
+  sendRequestUsingCrossDomainIframe(request, transportOptions) {
+    const frameData = Transport.crossDomainIframes_[transportOptions['iframe']];
+    user().assert(frameData, 'Trying to send message to non-existent frame');
+    frameData.msgQueue.push({senderId: this.id_, request});
+    if (frameData.isReady) {
+      if (frameData.sendTimer == null) {
+        frameData.sendTimer = timerFor(window).delay(() => {
+          Transport.sendQueuedMessagesToCrossDomainIframe_(frameData);
+          frameData.sendTimer = null;
+        }, MESSAGE_THROTTLE_TIME_);
+      }
+    }
+  }
+
+  /**
+   * If iframe is specified in config/transport, check whether third-party
+   * iframe already exists, and if not, create it.
+   * @param {!HTMLDocument} ampDoc The AMP document
+   * @param {!Object<string,string>} transportOptions The 'transport' portion
+   * of the amp-analytics config object
+   * @param {?Function<String>=} processResponse An optional function to
+   * receive any response messages back from the cross-domain iframe
+   */
+  processCrossDomainIframe(ampDoc, transportOptions, processResponse) {
+    user().assert(transportOptions['iframe'],
+      'Cross-domain frame parameters missing');
+    const frameUrl = transportOptions['iframe'];
+    // If iframe doesn't exist for this iframe url, create it.
+    if (Transport.hasCrossDomainFrame(frameUrl)) {
+      Transport.incrementFrameUsageCount_(frameUrl);
+      this.sendExtraData(frameUrl, transportOptions['extraData']);
+    } else {
+      const frame = this.createCrossDomainIframe_(ampDoc, frameUrl,
+        transportOptions['extraData']);
+      ampDoc.body.appendChild(frame);
+    }
+    // Regardless of whether we just created it, or are re-using an existing
+    // one, wire up the response callback
+    if (processResponse) {
+      const imc = Transport.crossDomainIframes_[frameUrl].iframeMessagingClient;
```

Comment:
Yeah, I also dislike the current name. The line is precisely 80 chars, hence choosing something with length=3. Will switch to better name and just bite the bullet and carriage-return on "."